.github/workflows/pytest.yaml

```diff
@@ -0,0 +1,69 @@
+name: Run mypy and pytest
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'release/*'
+    tags:
+      - 'v*'
+  pull_request:
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python-version: [3.7, 3.8]
+    env:
+      DISPLAY: ':99.0'
+      OS: ${{ matrix.os }}
+      PYTHON: ${{ matrix.python-version }}
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: '0'
+    - name: setup ubuntu-latest xvfb
+      uses: ./.github/actions/setup-ubuntu-latest-xvfb
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: prepare pip cache
+      id: prepare-cache-pip
+      uses: ./.github/actions/prepare-cache-pip
+    - name: pip cache
+      uses: actions/cache@v2
+      with:
+        path: ${{ steps.prepare-cache-pip.outputs.cache-dir }}
+        key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}-${{ hashFiles('**/test_requirements.txt') }}-${{ hashFiles('**/setup.py') }}
+        restore-keys: |
+          ${{ runner.os }}-pip-
+    - name: install requirements.txt
+      run: |
+        pip install -r requirements.txt --use-feature=2020-resolver
+    - name: install test_requirements.txt
+      run: |
+        pip install -r test_requirements.txt --use-feature=2020-resolver
+    - name: generate test dbs
+      uses: ./.github/actions/generate-test-dbs
+    - name: install qcodes
+      run: pip install . --use-feature=2020-resolver
+    - name: Run Mypy
+      run: mypy qcodes
```

Comment:
i haven't read about actions enough, hence a quick question - if mypy step fails, will the subsequent test steps be executed?

also, what do we want? i remember myself prefering to run tests despite mypy problems being useful for getting an overview of errors which is potentially useful for reducing number of pushes and hence ci triggers ))

Reply:
As far as I understand it will terminate immediately.

Reply:
ok, we can look at it later if needed